modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1263,6 +1263,43 @@ void ONNXImporter::handleNode(const opencv_onnx::NodeProto& node_proto_)
             }
             int outCn = layerParams.blobs.empty() ? outShapes[node_proto.input(1)][0] : layerParams.blobs[0].size[0];
             layerParams.set("num_output", outCn);
+
+            // Check for asymmetric padding
+            if (layerParams.has("pad"))
+            {
+                bool asymmetricPadding = false;
+                DictValue pads = layerParams.get("pad");
+                const int dims = pads.size() / 2;
+                for (int i = 0; i < dims; ++i)
+                {
+                    if (pads.get<int>(i) != pads.get<int>(i + dims))
+                    {
+                        asymmetricPadding = true;
+                        break;
+                    }
+                }
+                if (asymmetricPadding)
+                {
+                    layerParams.erase("pad");
+                    std::vector<int> paddings(pads.size());
```

Comment:
Please add a comment that this line ends up not padding N and C out of NCHW.

Reply:
Isn't padding always done along H and W dimensions in convolution layer?

Reply:
I got confused looking at this line and thought there is an error and opened a debugger to understand why it's done this way. Maybe it's just me.

Reply:
Added comments in latest commit. Please check

Reply:
Looks good, thank you!